packages/headless/src/features/triggers/trigger-analytics-actions.ts

```diff
@@ -0,0 +1,15 @@
+import {AnalyticsType, makeAnalyticsAction} from '../analytics/analytics-utils';
+
+/**
+ * Log trigger query
+ */
+export const logQueryTrigger = makeAnalyticsAction(
+  'analytics/trigger/query',
+  AnalyticsType.Search,
+  (client, state) => {
+    if (state.triggers && state.triggers.query !== null) {
```

Comment:
```suggestion
    if (state.triggers?.query) {
```

should be sufficient :)